components/api-server/src/routes.rs

```diff
@@ -0,0 +1,170 @@
+use axum::{
+    Json,
+    extract::{Path, State},
+    http::StatusCode,
+    response::{
+        IntoResponse,
+        Sse,
+        sse::{Event, KeepAlive},
+    },
+    routing::get,
+};
+use futures::{Stream, StreamExt};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use tower_http::cors::{Any, CorsLayer};
+use utoipa::{OpenApi, ToSchema};
+use utoipa_axum::{router::OpenApiRouter, routes};
+
+use crate::client::{Client, ClientError, QueryConfig};
+
+// `utoipa::OpenApi` triggers `clippy::needless_for_each`
+#[allow(clippy::needless_for_each)]
+mod api_doc {
+    #[derive(utoipa::OpenApi)]
+    #[openapi(info(
+        title = "API Server",
+        description = "API Server for CLP",
+        contact(name = "YScope")
+    ))]
+    pub struct ApiDoc;
+}
+use api_doc::ApiDoc;
+
+#[utoipa::path(
+    get,
+    path = "/health",
+    responses((status = OK, body = String))
+)]
+async fn health() -> String {
+    "API server is running".to_owned()
+}
+
+#[utoipa::path(
+    post,
+    path = "/query",
+    description = "Submits a new query job.",
+    request_body(
+        content= QueryConfig,
+        example = json!({
+            "query_string": "*",
+            "dataset": "default",
+            "begin_timestamp": 0,
+            "end_timestamp": 17_356_896,
+            "ignore_case": true,
+            "max_num_results": 0,
+            "write_to_file": false
+        })),
+    responses(
+        (
+            status = OK,
+            body = QueryResultsUri,
+            example = json!({"query_results_uri":"/query_results/1"})
+        ),
+        (status = INTERNAL_SERVER_ERROR)
+    )
+)]
+async fn query(
+    State(client): State<Client>,
+    Json(query_config): Json<QueryConfig>,
+) -> Result<Json<QueryResultsUri>, HandlerError> {
+    let search_job_id = client.submit_query(query_config).await?;
+    let uri = format!("/query_results/{search_job_id}");
+    Ok(Json(QueryResultsUri {
+        query_results_uri: uri,
+    }))
+}
+
+#[derive(Clone, Serialize, Deserialize, ToSchema)]
+#[serde(deny_unknown_fields)]
+struct QueryResultsUri {
+    /// The uri to get the query results.
+    query_results_uri: String,
+}
+
+#[utoipa::path(
+    get,
+    path = "/query_results/{search_job_id}",
+    description = "Streams the results of a previously submitted query as Server-Sent Events \
+        (SSE).",
+    responses(
+        (
+            status = OK,
+            body = String,
+            content_type = "text/event-stream",
+            description = "Server-Sent Events stream of query results. Each event contains a \
+                single line of the query result in JSON format.",
```

Comment:
```suggestion
                single line of the query result in JSON format.",
            example = r#"data: {"timestamp": 1633036800, "message": "Example log message"}\n"#
```
Add a short example to address my own notes last night.